pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -520,6 +518,41 @@ func (cd *clusterDeploy) cacheAgentImages(name string) error {
 	return nil
 }
 
+func (cd *clusterDeploy) getPrivateRegistrySecret(cluster *v3.Cluster) (map[string][]byte, error) {
```

Comment:
nit: We should probably add a comment outlining the logic in summary form, and maybe call it `getPrivateRegistrySecretData` as it's returning data rather than the secret itself.